### PR TITLE
Issue/#22 implement user input validation

### LINKED ISSUE #22: Implement User Input Validation
As we interact with users, it's crucial to validate the user input to ensure that it's in the expected format and within the expected range. Incorrect user input can lead to bugs, errors, or even security vulnerabilities. In this task, we'll create a user input validation system for our application.

Tasks include:

- **Identify User Input**: Identify all areas in our application where we accept user input. This could include text inputs, file uploads, and form submissions.
- **Define Validation Rules**: For each type of input, define what constitutes valid and invalid input. This could include input length, format, range, etc.
- **Implement Validation**: Implement validation checks for each type of user input in our application. This should happen before any processing or manipulation of the input data.
- **Error Handling**: Ensure that our application handles invalid inputs gracefully. This might involve displaying an error message to the user or requesting them to enter the data again.
- **Testing**: Test the validation system thoroughly with different types of input, including edge cases.

**Acceptance Criteria**:

1. All user input in the application is validated according to defined rules.
2. The application handles invalid inputs gracefully, without crashing or causing bugs.
3. Validation system has been tested and is working as expected.

### PR DESCRIPTION
**Pull request by FreakyF**
Closes #22

### CODE REVIEW

File: src/main/java/org/weathertrack/service/validation/UserInputValidator.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.weathertrack.service.validation.model.InputValidationResult;
+import org.weathertrack.service.validation.util.ValidationMessage;
+
+import java.util.regex.Pattern;
+
+public class UserInputValidator implements InputValidator {
+	private static final int MAXIMUM_CITY_NAME_LENGTH = 170;
+
+	private static final Pattern CITY_NAME_PATTERN_NUMBERS = Pattern.compile(".*\\d+.*");
+	private static final Pattern CITY_NAME_PATTERN_SPECIAL_CHARACTERS = Pattern.compile(".*[^\\p{L}-'\\d\\s]+.*");
```
Comment (Tymisko) [inline src/main/java/org/weathertrack/service/validation/UserInputValidator.java]: Write a comment describing this pattern.

**Review by Tymisko — CHANGES_REQUESTED** [review]

**Review by Tymisko — APPROVED** [review]
